### PR TITLE
Update JGit version

### PR DESCRIPTION
**Pull request by octoshikari**
In new 6.5 version PRESERVE option has been replaced by MERGES. Please check release notes:
https://projects.eclipse.org/projects/technology.jgit/releases/6.5.0

Also support new setDepth option in git-clone  got shallow clone

Related issue:
- https://github.com/clj-jgit/clj-jgit/issues/80

@edannenberg check please when you will have time. Cheers

### CODE REVIEW

**Comment by octoshikari** [comment]
@edannenberg For automatically run workflow you should  allow all actions for any user. Please see screenshot below
Снимок экрана 2023-06-22 в 01 04 06

**Comment by edannenberg** [comment]
It was already on but first-time contributers required approval. Thanks for pointing me in right direction.

The JGit June release dropped last week, changelog looks like it should be a uneventful bump. might as well use the latest and greatest.

**Comment by octoshikari** [comment]
@edannenberg Hey! I have updated jgit version to latest version 6.6.0. I also made a small change to the workflow file. Now clj-kondo is installed with lein in 1 step and lint step will only fail on errors. I can fix all lint warnings if you like in this PR or in a separate

File: .github/workflows/tests-on-push.yml
Comment (edannenberg) [inline .github/workflows/tests-on-push.yml:26]: Are double quotes handled differently?

File: .github/workflows/tests-on-push.yml
Comment (edannenberg) [inline .github/workflows/tests-on-push.yml:57]: I fixed the remaining linter warnings, so we can drop the `--fail-level` part.

**Review by edannenberg — CHANGES_REQUESTED** [review]

File: .github/workflows/tests-on-push.yml
Comment (octoshikari) [reply .github/workflows/tests-on-push.yml]: It handled in same way. Formatter change it to double quotes. Anyway i've returned single quotes

**Comment by edannenberg** [comment]
Thank you! `1.1.0-SNAPSHOT` is up on clojars. Gonna wait a bit with the next stable release until the diffing/git-diff PR is merged. Feel free to poke me should I get side-tracked by other stuff again.

**Pull request merged**
